sql/core/src/test/scala/org/apache/spark/sql/CollationRegexpExpressionsSuite.scala

```diff
@@ -34,288 +34,380 @@ class CollationRegexpExpressionsSuite
     // Supported collations
     case class LikeTestCase[R](l: String, r: String, c: String, result: R)
     val testCases = Seq(
-      LikeTestCase("ABC", "%B%", "UTF8_BINARY", true)
+      LikeTestCase("ABC", "%B%", "UTF8_BINARY", true),
+      LikeTestCase("AḂC", "%ḃ%", "UTF8_BINARY_LCASE", true), // scalastyle:ignore
+      LikeTestCase("ABC", "%b%", "UNICODE", false)
     )
     testCases.foreach(t => {
-      val query = s"SELECT like(collate('${t.l}', '${t.c}'), collate('${t.r}', '${t.c}'))"
+      val query = s"SELECT like(collate('${t.l}', '${t.c}'), '${t.r}')"
       // Result & data type
       checkAnswer(sql(query), Row(t.result))
       assert(sql(query).schema.fields.head.dataType.sameType(BooleanType))
-      // TODO: Implicit casting (not currently supported)
     })
     // Unsupported collations
     case class LikeTestFail(l: String, r: String, c: String)
     val failCases = Seq(
-      LikeTestFail("ABC", "%b%", "UTF8_BINARY_LCASE"),
-      LikeTestFail("ABC", "%B%", "UNICODE"),
       LikeTestFail("ABC", "%b%", "UNICODE_CI")
     )
     failCases.foreach(t => {
-      val query = s"SELECT like(collate('${t.l}', '${t.c}'), collate('${t.r}', '${t.c}'))"
+      val query = s"SELECT like(collate('${t.l}', '${t.c}'), '${t.r}')"
       val unsupportedCollation = intercept[AnalysisException] { sql(query) }
       assert(unsupportedCollation.getErrorClass === "DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE")
     })
-    // TODO: Collation mismatch (not currently supported)
   }
 
   test("Support ILike string expression with collation") {
     // Supported collations
     case class ILikeTestCase[R](l: String, r: String, c: String, result: R)
     val testCases = Seq(
-      ILikeTestCase("ABC", "%b%", "UTF8_BINARY", true)
+      ILikeTestCase("ABC", "%b%", "UTF8_BINARY", true),
+      ILikeTestCase("AḂC", "%ḃ%", "UTF8_BINARY_LCASE", true), // scalastyle:ignore
```

Comment:
Let's wrap the entire class with ascii check ignore such as in https://github.com/apache/spark/blob/6232085227ee2cc4e831996a1ac84c27868a1595/sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala#L27